### ISSUE TITLE
IPT collection_cde mapping recently changed and seems wrong

### ISSUE THREAD

**Issue by mkoo** [ArctosDB/arctos#4262]
A call with Doug Boyer and Julie Winchester highlighted recent changes to the Arctos IPT that make little sense.

See example:
https://www.idigbio.org/portal/records/f96066f3-d60d-4f4e-b78a-62bfb0f144a5
**Screenshot**
![Firefox_Screenshot_2022-01-13T22-08-29 462Z](https://user-images.githubusercontent.com/2523089/149418129-b7099078-2b48-439b-b461-d43fb18ec33f.png)

- Institution Code: ```Mvz``` despite weird title case, this is correct
- Collection Code: should not be this URL to the collection (that can be provided in Recordset section) and should be ```herp```
- catalog number: currently is ```mvz:herp:257608``` which is actually the entire DarwinCore triple or GUID, not really the catalog number

Basically the issue is that the DwC triple cannot be reconstructed from these three fields any longer.
If we want to have GUID included seems that needs to be an IPT change not handled like this

(occurrenceID provides proper URI to Arctos record)

**Comment by dustymc** [comment]
> the issue is that the DwC triple cannot be reconstructed from these three fields 

That's been true for a very long time - eg there are three "UAM:Herb" collections under that formula. (That's just within Arctos, unrealistic expectations get more unrealistic as the globe grows....)

GBIF has been making noises about actually sharing data (which is readily available from Arctos) - that requires an actionable collection identifier.

>catalog number

https://github.com/ArctosDB/arctos/issues/1781


>occurrenceID provides proper URI 

That it does, but it's also not a particularly stable identifier, just in case that was going to become the next issue!

How can we come up with a plan? I don't think arbitrarily flipflopping things around is in anyone's best interest.

**Comment by ebraker** [comment]
I just noticed this today for our MorphoSource records (populated via iDigBio API). The issue is fairly recent, as my previous MS batches as of October 1 and earlier look fine. This is how the title of each media file displays:


![image](https://user-images.githubusercontent.com/11336485/149570798-1cedd1cd-2b91-49ba-8278-8e5ee67a97ee.png)


...which appears to be a combination of collection code: "http://arctos.database.museum/collection/UCM:Herp" and catalog number: "UCM:Herp:49665"

**Comment by Jegelewicz** [comment]
I think this is all because of https://github.com/ArctosDB/arctos/issues/3953#issuecomment-926965926

**Comment by Jegelewicz** [comment]
See also - https://github.com/ArctosDB/arctos/issues/3953#issuecomment-930571760

**Comment by mkoo** [comment]
Yes these are not arbitrary choices!
its Arctos IPT not being DwC compliant when we put as our collection code a URL and our entire GUID in catalog number. 
And yes this is a recent change that was bad (that change was arbitrary!) 

So we should revert, and if we want to take it up with the inadequacies of these fields then engaging with the community on DwC upgrades are in order.

**Comment by Jegelewicz** [comment]
FWIW I think that the collection code url should go in collectionID - https://dwc.tdwg.org/terms/#dwc:collectionID.

As it stands right now, there really is no good place for our catalog record url. I hope this will get remedied once the TDWG materialSample Task Group gets its work done....

**Comment by Jegelewicz** [comment]
Actually - here was my full take https://github.com/ArctosDB/arctos/issues/3953#issuecomment-930571760

NMMNH:Paleo as example

[Institution Code](https://dwc.tdwg.org/terms/#dwc:institutionCode) - NMMNHS
[Institution ID](url) - https://www.gbif.org/grscicoll/institution/bcc1478b-1409-43c3-a013-69586aa98753
[Collection Code](https://dwc.tdwg.org/terms/#dwc:collectionCode) - NMMNH:Paleo
[Collection ID](https://dwc.tdwg.org/terms/#dwc:collectionID) - https://arctos.database.museum/collection/NMMNH:Paleo

**Comment by dustymc** [comment]
>DwC compliant 

It is - as with any good exchange standard it's hard not to be - it's just not meeting unjustified expectations.

>collectionID

I don't disagree, but I think that would also require significant work from Dave.

ANYWAY - https://github.com/ArctosDB/PG_DDL/blob/master/shared_data/dwc_occurrence.sql - tell me what to put where...

(And https://github.com/ArctosDB/arctos/issues/2769 would be handy, again.)

**Comment by Jegelewicz** [comment]
> I think that would also require significant work from Dave.

This would be a good chance to test that theory. @dbloom and I talked about this sort of thing and it might not be as much work as it used to. Can we test if for a single collection?

**Comment by dustymc** [comment]
>single collection?

Not in the current architecture.

**Comment by Jegelewicz** [comment]
I mean, we could take the Arctos DWC output, add this one thing and then have Dave try to import....

**Comment by mkoo** [comment]
I dont think we need to test. We need to fix. The values in the current fields are just wrong.
Here's another downstream example when you try to use the GBIF API:
<img width="505" alt="Screen Shot 2022-01-16 at 3 10 02 PM" src="https://user-images.githubusercontent.com/2523089/149682038-5ea2590c-692f-4133-b71b-3d63e98cc28a.png">

GUID used to concatenate correctly as MVZ:Herp:134634 but now is confusing and wrong.

Code vs ID is the key. If we cannot include institution ID, collection ID then just have the Institution Code and collection Code go back to what we were using just recently

- Institution Code =MVZ
- Collection Code =Herp
- Catalog Number =123123

Changing the priority because I'm starting to get complaints from users (so it's not just me...)

**Comment by mkoo** [comment]
What is the status of fixing this? (ignoring institution id and collection id vis a vis: #3953)

-Institution cde= InstitutionCode (fine)
-COLLECTION_CDE= collectionCode

I dont know why the dwc_occurrence.sql  has this at line 177:
``` 
http://arctos.database.museum/collection/' || filtered_flat.guid_prefix as collectionCode,
```
this was changed two weeks ago so please restore previous!

**Comment by dustymc** [comment]
> status of fixing 

I'd still like to find some way of having a conversation about this. It's crazy to not share the information that would let GBIF DO STUFF, I'm not terribly comfortable ignoring Neal's advice, making a bunch of collections less-unique seems like it sorta borders on evil from here (minimally, it'll guarantee that we'll never be able to find them in Morphosource), and I really don't want to dump any surprises in Dave's lap.

If that's not possible - I have no idea really, I guess I'll just change whatever anyone asks me to change??

>two weeks

September.

**Comment by Jegelewicz** [comment]
We need to share it IN THE RIGHT PLACE - PLEASE see https://github.com/ArctosDB/arctos/issues/4262#issuecomment-1013500603

We need to go back to the way things were, then work with @dbloom to do things properly.

**Comment by mkoo** [comment]
AGREED-- I am not opposed to having that conversation AFTER this issue is
fixed. Basically without discussion, the field value was changed for
collection code and now we have broken links downstream since no one can
create DwC triples from our IPT. We have shown two examples but I can show
more (Morphosource, AmphibiaWeb)....

Teresa has shown that the correct fields are the _id fields, which can be
adjusted with Dave once we set up a system for proper compensation

**Comment by dustymc** [comment]
``` 
 filtered_flat.collection_cde as collectionCode,
```

is running, generally takes about 5 hours to rebuild.
